### PR TITLE
Fix xcm erc20

### PR DESCRIPTION
**Pull request by zjb0807**
The `erc20_holding_account`(`0x000000000000000000ff00000000000000000000`, `5EMjsczLB8VLkyWNp8bRHhzvfGLe7DfPPRDHisw5WKVtokbi `) needs an initial balance and needs to monitor the minimum balance.

### CODE REVIEW

File: modules/currencies/src/mock.rs [hunk 1]
```diff
@@ -205,6 +205,8 @@ impl Config for Runtime {
 	type WeightInfo = ();
 	type AddressMapping = MockAddressMapping;
 	type EVMBridge = module_evm_bridge::EVMBridge<Runtime>;
+	type PaymentTransfer = ();
+	type StorageDepositFee = ConstU128<0>;
```
Comment (shaunxw) [inline modules/currencies/src/mock.rs:209]: unit tests needed to make sure `PaymentTransfer` and `StorageDepositFee` work as expected

File: modules/support/src/lib.rs [hunk 2]
```diff
@@ -96,6 +96,10 @@ pub trait TransactionPayment<AccountId, Balance, NegativeImbalance> {
 	fn apply_multiplier_to_fee(fee: Balance, multiplier: Option<Multiplier>) -> Balance;
 }
 
+pub trait PaymentTransfer<AccountId, Balance> {
+	fn payment_transfer(from: &AccountId, to: &AccountId, amount: Balance) -> DispatchResult;
+}
```
Comment (shaunxw) [inline modules/support/src/lib.rs:101]: need docstring

File: runtime/acala/src/lib.rs [hunk 3]
```diff
@@ -849,6 +849,8 @@ parameter_types! {
 	pub const GetLiquidCurrencyId: CurrencyId = LDOT;
 	pub const GetStakingCurrencyId: CurrencyId = DOT;
 	pub Erc20HoldingAccount: H160 = primitives::evm::ERC20_HOLDING_ACCOUNT;
+			// StorageDepositPerByte is 18 decimals, convert to 12 decimals.
+			pub StorageDepositFee: Balance = <StorageDepositPerByte as Get<Balance>>::get().saturating_div(1000000).saturating_mul(erc20::TRANSFER.storage.into());
```
Comment (shaunxw) [inline runtime/acala/src/lib.rs:853]:
```suggestion
			pub StorageDepositFee: Balance = <StorageDepositPerByte as Get<Balance>>::get().saturating_div(1_000_000).saturating_mul(erc20::TRANSFER.storage.into());
```

File: runtime/integration-tests/src/relaychain/erc20.rs [hunk 4]
```diff
@@ -306,6 +308,11 @@ fn erc20_transfer_between_sibling() {
 	});
 }
 
+#[test]
+fn erc20_transfer_between_sibling_pay_with_non_native_token() {
+	// TODO
+}
```
Comment (shaunxw) [inline runtime/integration-tests/src/relaychain/erc20.rs:314]: todo

File: runtime/mandala/src/lib.rs [hunk 5]
```diff
@@ -908,6 +908,8 @@ parameter_types! {
 	pub const GetNativeCurrencyId: CurrencyId = ACA;
 	pub const GetStableCurrencyId: CurrencyId = AUSD;
 	pub Erc20HoldingAccount: H160 = primitives::evm::ERC20_HOLDING_ACCOUNT;
+	// StorageDepositPerByte is 18 decimals, convert to 12 decimals.
+	pub StorageDepositFee: Balance = <StorageDepositPerByte as Get<Balance>>::get().saturating_div(1000000).saturating_mul(erc20::TRANSFER.storage.into());
```
Comment (shaunxw) [inline runtime/mandala/src/lib.rs:912]:
```suggestion
	pub StorageDepositFee: Balance = <StorageDepositPerByte as Get<Balance>>::get().saturating_div(1_000_000).saturating_mul(erc20::TRANSFER.storage.into());
```

**Comment by zjb0807** [comment]
Replaced by #2454

**Pull request closed without merging**